cellxgene_resources/curation_qa.ipynb

```diff
@@ -656,6 +656,133 @@
+    "nonpar_adata = raw_adata[:,raw_adata.var.feature_name.isin(nonpar['nonpar_genes'])] \n",
+    "region_dict = {'par': [par_adata], 'nonpar': [nonpar_adata]}\n",
+    "\n",
+    "#Check to see how many par/nonpar genes were found\n",
+    "print(f'% PAR genes found: {(par_adata.shape[1]/len(par[\"par_genes\"]))*100}')\n",
+    "print(f'% NonPAR genes found: {(nonpar_adata.shape[1]/len(nonpar[\"nonpar_genes\"]))*100}')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Check for batch or library uuid fields for correct grouping\n",
+    "print(raw_adata.obs_keys)\n",
+    "library_id = 'batch'  # this variable will change depending on the dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Construct new expression dataframes for par and nonpar\n",
+    "for k,v in region_dict.items():\n",
+    "    a = v[0]\n",
+    "    df = pd.DataFrame()\n",
+    "    for gene in a.var.index:\n",
+    "        a.obs[gene] = a[:,a.var.index == gene].X[:,0].toarray()  # add each gene exp counts to obs\n",
+    "        df[gene] = a.obs.groupby(['donor_id', library_id])[gene].sum()  # sum counts of each gene by donor and library id\n",
+    "        v.append(df)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Sum expression counts across regions\n",
+    "par_df = region_dict['par'][1]\n",
+    "par_df['par_sum'] = np.sum(par_df, axis=1)\n",
+    "nonpar_df = region_dict['nonpar'][1]\n",
+    "nonpar_df['nonpar_sum'] = np.sum(nonpar_df, axis=1)\n",
+    "nonpar_par = pd.merge(nonpar_df, par_df, how='left', left_index=True, right_index=True) \n",
+    "nonpar_par['nonPAR_to_PAR'] = nonpar_par['nonpar_sum']/nonpar_par['par_sum']  # calculate nonpar-to-par ratio"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Generate donor sex summary dataframe\n",
+    "nonpar_par.reset_index(inplace=True)\n",
+    "counts_per_lib = nonpar_par.value_counts(['donor_id',library_id,'nonPAR_to_PAR']).reset_index().rename(columns={\"index\": \"value\", 0: \"count\"}) #this drops all na values\n",
+    "counts_per_lib.drop(columns=['count'],inplace=True)\n",
+    "counts_per_lib.set_index(['donor_id'], inplace=True)\n",
+    "counts_avg = counts_per_lib.groupby(level='donor_id').mean()  # collapse multi-index, calculating average expression counts across donor libraries/batches\n",
+    "counts_avg.reset_index(inplace=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Assign sex using 0.5 threshold\n",
+    "def assign_sex(x):\n",
+    "    if x >= 0.5:\n",
+    "        return 'Male'\n",
+    "    elif x < 0.5:\n",
```

Comment:
We talked about not having 0.5 be a strict cut-off but rather have narrower ranges for Male & Female with a gap in the middle, then keep an eye out for any that fall into the gap to see if we should adjust. I think we said keep male 0.5 and over, but female would be < 0.2